src/release/breaking-changes/boxdecoration-shape-migration.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Migration guide for BoxDecoration.circle
+description: Removal of BoxDecoration.circle and how to migrate
+---
+
+## Summary
+
+This change removes `BoxDecoration(shape: BoxShape.circle)`. Existing apps
+should use `ShapeDecoration(shape: CircleBorder())` instead.
+
+## Context
+
+`BoxDecoration(shape: BoxShape.circle)` uses an enum. Each new enum value requires new code
+in many places. That new code may not shake out,
+since the compiler doesn't know which values never get used.
+`BoxDecoration` has a permanent binary size cost,
+even for people who don't use it.
+Flutter moved to `ShapeBorders` rather than `BoxShape` for new
+shapes, and the existence of `BoxShape` remains as a legacy feature.
+
+Use `ShapeDecoration`. It aligns with the rest of the Flutter API.
+`BoxShape` has been deprecated.
+Continuing to use that feature can cause confusion.
+
+## Description of change
+
+Remove `BoxDecoration(shape: BoxShape.circle)` and `BoxDecoration.shape`.
+
+## Migration guide
+
+- Replace `BoxDecoration(shape: BoxShape.circle)` with
+`ShapeDecoration(shape: CircleBorder())`.
+- Remove the `BoxDecoration.shape` property from
+`BoxDecoration(shape: BoxShape.rectangle)`.
+
+When you use `Border`,
+`CircleBorder(width: ..., color: ..., strokeAlign: ...)` inherits
+the `Border.all(width: ..., color: ..., strokeAlign: ...)` values.
```

Comment:
I am not sure where this fits into the context of this guide. Can you explain more?

Reply:
If you have `BoxDecoration(shape: BoxShape.circle, border: Border.all(width: 20)`, this becomes now `ShapeDecoration(shape: CircleBorder(side: BorderSide(width: 20)))`. In this scenario, Border becomes BorderSide.